### ISSUE TITLE
Steam OS stuck at "Verifying Installation" after update [Report]

### ISSUE THREAD

**Issue by Jbrincklow** [holoiso-eol/holoiso#612]
## Your Issue:
```
I have the steam OS dual booted from an SSD. Was working great until steam decided to update. It told me it was going to drop me into steam and had the boot logo and startup sound, but then it froze. On reboot, it has verifying Installation with a grey (empty) loading bar. Same start up sound, but still frozen.


```

## Your device's full specs
```
CPU: 5600x
GPU: 6700xt
Memory: 32 gb 3200 mhz
Ssd: 500 gb Samsung 870
```

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
Unknown, how do I check without access to a terminal?
```

## (Only for SteamOS Session loading issues) Have you tried launching `gamescope-session` from terminal? If yes, provide a log.
Locked out of terminal
```
yeah pubg crash bhai
```

## Your installation type:
```
deckperience
```

## Have you updated your installation recently?
```
Yep, not by choice
```

**Comment by Salad360** [comment]
FYI - if you are stuck at "Verifying installation" screen you should be able to drop to a terminal session by hitting CTRL + ALT + F4

**Comment by AaronBaird1** [comment]
press CTRL+ALT+F2 and then type:
`steamos-session-select plasma`
`reboot`

It will reboot into desktop.  still waiting on a fix for game mode.

**Comment by Jbrincklow** [comment]
Ok, made it. I don't remember my user login, but I do remember my root login. When I type that line in I get

`usr/bin/sed: -e expression #1, char 0: no previous regular expression`

Then of course the reboot still works

**Comment by Jbrincklow** [comment]
Reset my user password.
SteamOS version snapshot20220921.1733

![16689888633636046020184447716760](https://user-images.githubusercontent.com/24421822/202933818-8d11145d-876f-4be2-b9b6-e3801c80845a.jpg)

**Comment by Jbrincklow** [comment]
Fixed it.

`Sudo vim /etc/sddm.conf.d/autologin.conf`

[Autologin]
User="YourUsername"
Session=plasma
This configuration causes a KDE Plasma

Can't use game mode as mentioned, but don't really need it. Now I can go back to playing EA games that don't work on Windows for some ridiculous reason.

**Comment by AaronBaird1** [comment]
why did you close it? gamemode still needs to be fixed.

**Comment by Jbrincklow** [comment]
Reopened for gamemode fix

**Comment by moodsinmotion** [comment]
> Ok, made it. I don't remember my user login, but I do remember my root login. When I type that line in I get
> 
> `usr/bin/sed: -e expression #1, char 0: no previous regular expression`
> 
> Then of course the reboot still works

Having the same problem here. Wanna try to fix this "verifying installation" issue and have forgotten my login data. How did you manage to work around the login? thx for helping me out

**Comment by theVakhovskeIsTaken** [comment]
Unable to reproduce on all of my devices.

Are you able to run gamescope-session in desktop mode fine?

**Comment by hh1599** [comment]
@theVakhovskeIsTaken 

I also have a Win Max 2 device and based on reports of game mode booting on external displays but freezing on the internal display i think it is a problem with the display somewhere.  

This is the result of `gamescope-session` in desktop mode.

> gamescope-session
no Intel GPU, skipping color compression disabling
Making Gamescope Mode Save file at "/home/ab/.config/gamescope/modes.cfg"
Claimed global gamescope stats session at "/run/user/1000/gamescope-stats"
wlserver: [backend/headless/backend.c:82] Creating headless backend
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
vulkan: selecting physical device 'AMD RADV YELLOW_CARP': queue family 1
vulkan: physical device supports DRM format modifiers
vulkan: supported DRM formats for sampling usage:
vulkan:   0x34325241
vulkan:   0x34325258
vulkan:   0x3231564E
wlserver: Running compositor on wayland display 'gamescope-0'
wlserver: [backend/headless/backend.c:18] Starting headless backend
wlserver: [xwayland/sockets.c:63] Failed to bind socket @/tmp/.X11-unix/X0: Address already in use
wlserver: [xwayland/sockets.c:63] Failed to bind socket @/tmp/.X11-unix/X0: Address already in use
wlserver: [xwayland/server.c:92] Starting Xwayland on :1
wlserver: [xwayland/server.c:92] Starting Xwayland on :2
wlserver: [types/wlr_surface.c:741] New wlr_surface 0x557c5b7ce510 (res 0x557c5c0dac60)
wlserver: [xwayland/server.c:250] Xserver is ready
wlserver: [types/wlr_surface.c:741] New wlr_surface 0x557c5b7ce850 (res 0x557c5ad225d0)
wlserver: [xwayland/server.c:250] Xserver is ready
pipewire: stream state changed: connecting
pipewire: stream state changed: paused
pipewire: stream available on node ID: 66
pipewire: renegotiating stream params (size: 1280x720)
Error : /etc/xbindkeysrc not found or reading not allowed.
please, create one with 'xbindkeys --defaults > /etc/xbindkeysrc'.
or, if you want scheme configuration style,
with 'xbindkeys --defaults-guile > /home/ab/.xbindkeysrc.scm'.
/usr/bin/gamescope-session: line 257: /usr/bin/ibus-daemon: No such file or directory
/usr/bin/gamescope-session: line 273: /tmp/steamos-short-session-tracker: No such file or directory
wlserver: [types/wlr_surface.c:741] New wlr_surface 0x557c5c0d9900 (res 0x557c5b7d4d40)
[2022-11-22 11:10:23.443] [info] [config.cpp:117] parsing config: '/tmp/mangohud.wpJWYeC9'
[2022-11-22 11:10:23.463] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-HDMI-A-1/device/hwmon/': No such file or directory
[2022-11-22 11:10:23.463] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DP-6/device/hwmon/': No such file or directory
[2022-11-22 11:10:23.463] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-eDP-1/device/hwmon/': No such file or directory
[2022-11-22 11:10:23.463] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DP-4/device/hwmon/': No such file or directory
[2022-11-22 11:10:23.463] [error] [file_utils.cpp:43] Error opening directory '/sys/class/drm/card0-DP-2/device/hwmon/': No such file or directory
[2022-11-22 11:10:23.464] [info] [overlay.cpp:733] Uploading is disabled (permit_upload = 0)
xwm: Rejecting WM_CHANGE_STATE to ICONIC for window 0x400007
steam.sh[3362]: Running Steam on steamos rolling 64-bit
steam.sh[3362]: STEAM_RUNTIME is enabled automatically
setup.sh[3503]: Steam runtime environment up-to-date!
steam.sh[3362]: Steam client's requirements are satisfied
gamescope: received kill signal, terminating!
(EE) failed to read Wayland events: Broken pipe
(EE) failed to read Wayland events: Broken pipe
gamescope-session: unexpected background pid 3361 at teardown:
    PID TTY          TIME CMD
   3361 pts/1    00:00:00 mangoapp
/usr/bin/gamescope-session: line 335:  3361 Killed                  mangoapp
/usr/bin/gamescope-session: line 335:  3554 Killed                  sleep 5


Anything I can try?

**Comment by segaboy** [comment]
I've been stuck with this issue for almost a week. Has anyone been able to boot into game mode yet?

**Comment by cstruck** [comment]
this seems to be a bug if you've set a fixed resolution inside gamescope. The verify will work if you remove the `~/.config/gamescope/modes.cfg` sadly this means you have to set the resolution again after the next start. but it fixes the start of the gamescope session.

This was also already mentioned here https://github.com/theVakhovskeIsTaken/holoiso/issues/606#issuecomment-1321219923

**Comment by AaronBaird1** [comment]
> this seems to be a bug if you've set a fixed resolution inside gamescope. The verify will work if you remove the `~/.config/gamescope/modes.cfg` sadly this means you have to set the resolution again after the next start. but it fixes the start of the gamescope session.
> 
> This was also already mentioned here [#606 (comment)](https://github.com/theVakhovskeIsTaken/holoiso/issues/606#issuecomment-1321219923)

On the Min Max 2 this file is empty and deleting it does not solve the issue.

Perhaps there is a clue in these logs?
[gamescope](https://gist.github.com/hh1599/d372d01f6af58e87d31b2dd48a9aecec#file-gamescope)
[steam](https://gist.github.com/hh1599/d372d01f6af58e87d31b2dd48a9aecec#file-steam)

**Comment by kartsnkittiesGF** [comment]
I am also experiencing this issue on my Win Max 2. I installed successfully, was prompted to update upon first setup, went through with the update and then was unable to launch back into steam ui. Rebooted, and then stuck on Verifying installation. Was able to get into desktop mode using instructions above. Switching to game mode results in a black screen. I decided to try to reinstall, since it was working prior to updating through the Steam ui, and am seeing an error "cannot open display :0" however it seems the reinstall succeeded and is working so far. This is my first time installing or using HoloISO, so I'm not 100% sure of what to expect, or what is normal behavior and what isn't

**Comment by ssj4maiko** [comment]
I guess I saw this temp "solution" somewhere else, so I guess it's better to report here too just in case. Also a GPDWM2 user here.

I connected to an external monitor and I was able to boot into `gamescope`, I then removed the cable and everything ran normally. I went to desktop, everything was fine, but when I got back to Gaming Mode, the problem returned.

So as explained, seems to be a problem with resolution or something, although more specifically when booting gamescope, as it will work once you trigger an external monitor event just to force an update in that matter.

**Comment by kartsnkittiesGF** [comment]
I can also confirm that plugging into an external display and then unplugging will turn the built in display back on. Trying to see if I can find a workaround so as to not have to plug into a monitor after every reboot for the time being.

**Comment by rodti** [comment]
> #606 (comment)

Thanks, this got me out of a cycle of 'verifying installation'!

**Comment by Duckmanjbr** [comment]
> Fixed it.
> 
> `Sudo vim /etc/sddm.conf.d/autologin.conf`
> 
> [Autologin] User="YourUsername" Session=plasma This configuration causes a KDE Plasma
> 
> Can't use game mode as mentioned, but don't really need it. Now I can go back to playing EA games that don't work on Windows for some ridiculous reason.

How can we make this a permanent boot fix?  On boot I get hung at "Loading internal ramdisk....".  I can boot into plasma by doing the following:

press CTRL+ALT+F2 and then type:
```
steamos-session-select plasma
sudo reboot
```

However this needs to be done every time.  Updating the autologin.conf as stated above doesn't seem to fix the boot issue.  Is this a grub config that needs to be changed?

**Comment by ranenvious** [comment]
Is there any progress on the game-mode fix? I'm trying to retrofit an old desktop as a living room console and even if it's possible to sidestep this issue by using desktop mode that isn't really usable as a living room console.

**Comment by Jbrincklow** [comment]
> Is there any progress on the game-mode fix? I'm trying to retrofit an old desktop as a living room console and even if it's possible to sidestep this issue by using desktop mode that isn't really usable as a living room console.

Even when game-mode was usable, I could never get 4k output working, I was stuck with the steam deck's native resolution of 800p. You could probably do desktop mode with some freeware to enable mouse support on a controller. I myself use a wireless mouse and keyboard on a laptop table.

**Comment by ranenvious** [comment]
> > Is there any progress on the game-mode fix? I'm trying to retrofit an old desktop as a living room console and even if it's possible to sidestep this issue by using desktop mode that isn't really usable as a living room console.
> 
> Even when game-mode was usable, I could never get 4k output working, I was stuck with the steam deck's native resolution of 800p. You could probably do desktop mode with some freeware to enable mouse support on a controller. I myself use a wireless mouse and keyboard on a laptop table.

ok? Who said anything about 4k to begin with? upscaling to 1080p is pretty common to do with a dock and my TV is across the room. Even flagship consoles can't pull off 4k very well, I'm not sure why that's immediately where this conversation is heading.

**Comment by GKMoritz** [comment]
The gamescope problem seems to be fixed with 4.1, at least if you boot into Plasma first and then use the "back to gamemode" button.

Tested on 
Intel NUC6i7KYK
Intel Core 7-6770HQ
Intel Iris Pro 580
(has the issue since the first offline installer)
